auction/src/lib.rs

```diff
@@ -339,18 +507,28 @@ pub mod pallet {
 			Ok(().into())
 		}
 
-		#[pallet::weight(T::WeightInfo::remove_bid())]
+		#[pallet::weight((
+            {
+				let mut s:usize = 0;
+				let auction = Auctions::<T>::get(nft_id).ok_or(Error::<T>::AuctionDoesNotExist);
+				if let Ok(auction) = auction {
+					s = auction.bidders.list.len();
+				}
+				T::WeightInfo::remove_bid(s as u32)
+            },
+			DispatchClass::Normal
+        ))]
 		pub fn remove_bid(origin: OriginFor<T>, nft_id: NFTId) -> DispatchResultWithPostInfo {
 			let who = ensure_signed(origin)?;
 			let current_block = frame_system::Pallet::<T>::block_number();
 
-			// remove bid from storage
+			// remove bid from storage.
 			Auctions::<T>::try_mutate(nft_id, |maybe_auction| -> DispatchResult {
-				// should not panic when unwrap since already checked above
+				// should not panic when unwrap since already checked above.
 				let auction = maybe_auction.as_mut().ok_or(Error::<T>::AuctionDoesNotExist)?;
 
 				let remaining_blocks = auction.end_block.saturating_sub(current_block);
```

Comment:
I got the feeling that we need to move this kind of logic to the Auction Struct. This mean that we could benefit if the AuctionData has the following methods: `is_ending`, `is_started`, ....